model/config_test.go

```diff
@@ -840,19 +841,173 @@ func TestTagsEquals(t *testing.T) {
+		},
+			valid: false},
+		{name: "route rule bad weight dest", in: &proxyconfig.RouteRule{
+			Destination: "host.default.svc.cluster.local",
+			Match:       &proxyconfig.MatchCondition{Source: "somehost.default.svc.cluster.local"},
+			Route: []*proxyconfig.DestinationWeight{
+				{Destination: strings.Repeat(strings.Repeat("1234567890", 6)+".", 6)},
+			},
+		},
+			valid: false},
+		{name: "route rule bad weight", in: &proxyconfig.RouteRule{
+			Destination: "host.default.svc.cluster.local",
+			Match:       &proxyconfig.MatchCondition{Source: "somehost.default.svc.cluster.local"},
+			Route: []*proxyconfig.DestinationWeight{
+				{Weight: -1},
+			},
+		},
+			valid: false},
+		{name: "route rule bad route tags", in: &proxyconfig.RouteRule{
+			Destination: "host.default.svc.cluster.local",
+			Match:       &proxyconfig.MatchCondition{Source: "somehost.default.svc.cluster.local"},
+			Route: []*proxyconfig.DestinationWeight{
+				{Tags: map[string]string{"@": "~"}},
+			},
+		},
+			valid: false},
+		{name: "route rule bad timeout", in: &proxyconfig.RouteRule{
+			Destination: "host.default.svc.cluster.local",
+			HttpReqTimeout: &proxyconfig.HTTPTimeout{
+				TimeoutPolicy: &proxyconfig.HTTPTimeout_SimpleTimeout{
+					SimpleTimeout: &proxyconfig.HTTPTimeout_SimpleTimeoutPolicy{TimeoutSeconds: -1},
+				},
+			},
+		},
+			valid: false},
+		{name: "route rule bad retry attempts", in: &proxyconfig.RouteRule{
+			Destination: "host.default.svc.cluster.local",
+			HttpReqRetries: &proxyconfig.HTTPRetry{
+				RetryPolicy: &proxyconfig.HTTPRetry_SimpleRetry{
+					SimpleRetry: &proxyconfig.HTTPRetry_SimpleRetryPolicy{Attempts: -1},
+				},
+			},
+		},
+			valid: false},
+		{name: "route rule bad delay fixed seconds", in: &proxyconfig.RouteRule{
+			Destination: "host.default.svc.cluster.local",
+			HttpFault: &proxyconfig.HTTPFaultInjection{
+				Delay: &proxyconfig.HTTPFaultInjection_Delay{
+					Percent:       -1,
+					HttpDelayType: &proxyconfig.HTTPFaultInjection_Delay_FixedDelaySeconds{FixedDelaySeconds: -1},
+				},
+			},
+		},
+			valid: false},
+		{name: "route rule bad abort percent", in: &proxyconfig.RouteRule{
+			Destination: "host.default.svc.cluster.local",
+			HttpFault: &proxyconfig.HTTPFaultInjection{
+				Abort: &proxyconfig.HTTPFaultInjection_Abort{
+					Percent:   -1,
+					ErrorType: &proxyconfig.HTTPFaultInjection_Abort_HttpStatus{HttpStatus: 500},
+				},
+			},
+		},
+			valid: false},
+		{name: "route rule bad abort status", in: &proxyconfig.RouteRule{
+			Destination: "host.default.svc.cluster.local",
+			HttpFault: &proxyconfig.HTTPFaultInjection{
+				Abort: &proxyconfig.HTTPFaultInjection_Abort{
+					Percent:   100,
+					ErrorType: &proxyconfig.HTTPFaultInjection_Abort_HttpStatus{HttpStatus: -1},
+				},
+			},
+		},
+			valid: false},
+		{name: "route rule bad delay exp seconds", in: &proxyconfig.RouteRule{
+			Destination: "host.default.svc.cluster.local",
```

Comment:
You will see the `valid: false` a few lines down.  The test validates that the rule is invalid, as you expected.  Actually two errors are returned via multierror -- one error for the bad form, and an additional one for the fact that the feature is unimplemented.